gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobTask.java

```diff
@@ -203,6 +209,7 @@ public TaskResult run() {
       return new TaskResult(TaskResult.Status.FAILED, "Exception occurred for job " + planningJobId + ":" + ExceptionUtils
           .getFullStackTrace(e));
     } finally {
+      this.jobHelixManager.disconnect();
```

Comment:
formatting is a bit hard to discern (in the PR viewer), but is this the `finally` of the `try` block where the `.connect()` call is the first statement?  if so, shouldn't we either move the `.connect()` directly prior to the `try`, or else begin another `try` right after the `connect()`?